compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/input/internal/selection/TextFieldSelectionState.kt

```diff
@@ -407,6 +407,8 @@ internal class TextFieldSelectionState(
                         clearHandleDragging()
                     }
                 )
+            }.invokeOnCompletion {
+                clearHandleDragging()
```

Comment:
Are changes in `commonMain` necessary to fix the bug?

Reply:
If so, can you describe in details what this lines fixes?

(It needs to be upstreamed by us to AOSP if it fixes a critical issue)

Reply:
It is similar bug that was described [here](https://github.com/JetBrains/compose-multiplatform-core/pull/1000#issuecomment-2004723614) and [here](https://github.com/JetBrains/compose-multiplatform-core/pull/1000#issuecomment-2015584824).

But at this time in BTF2 it was addressed by Google and [workaround](https://github.com/JetBrains/compose-multiplatform-core/blob/714d64b01e1e508821b4317e22008ef6c5ef1c11/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/input/internal/selection/TextFieldSelectionState.kt#L612) was made. But because of [this check](https://github.com/JetBrains/compose-multiplatform-core/blob/714d64b01e1e508821b4317e22008ef6c5ef1c11/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/input/internal/selection/TextFieldSelectionState.kt#L605), there are still cases where magnifier can stuck (tap and hold selection handle -> press backspace -> release handle without moving it). 
Now it is technically happen because of this [listener](https://github.com/JetBrains/compose-multiplatform-core/blob/714d64b01e1e508821b4317e22008ef6c5ef1c11/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/input/internal/selection/TextFieldSelectionState.kt#L394C17-L394C39), drag listener is now working as designed

Reply:
Thanks! It is useful

cc @mazunin-v-jb, it will be needed to explain the fix in AOSP.

Reply:
This bug is not likely reproducible on Android without intentional programmatic selection manipulations or hardware keyboard usage. Software keyboard is disabled there while selection is performed